### ISSUE TITLE
Type mismatch false positive (dyn)

### ISSUE THREAD

**Issue by danielhuang** [rust-analyzer/rust-analyzer#11824]
With the following code:

```rs
fn test(x: Box<dyn Display>) {}

fn example() {
    test(Box::new(5))
}
```

rust-analyzer complains `expected Box<dyn Display, Global>, found Box<i32, Global>`, even though this code is valid.

Currently testing 

**rust-analyzer version**: 2022-03-25 dev

**rustc version**: not applicable

**relevant settings**: (eg. client settings, or environment variables like `CARGO`, `RUSTUP_HOME` or `CARGO_HOME`)

Experimental diagnostics enabled

**Comment by danielhuang** [comment]
This is a duplicate. Sorry, closing this issue.

**Comment by rkuhn** [comment]
for completeness: #11818
